Base/Python/slicer/util.py

```diff
@@ -2497,6 +2501,136 @@ def dataframeFromMarkups(markupsNode):
+                matrix3x3.SetElement(i_idx, j_idx, matrix4x4.GetElement(i_idx, j_idx))
+
+    # Get direction from volume node
+    directionMatrix4x4 = vtk.vtkMatrix4x4()
+    volumeNode.GetIJKToRASDirectionMatrix(directionMatrix4x4)
+    directionMatrix3x3 = vtk.vtkMatrix3x3()
+    _updateMatrix3x3From4x4(directionMatrix3x3, directionMatrix4x4)
+
+    # Transform from RAS (used in Slicer) to LPS (used in ITK)
+    rasToLps = vtk.vtkMatrix3x3()
+    rasToLps.SetElement(0, 0, -1)
+    rasToLps.SetElement(1, 1, -1)
+    vtk.vtkMatrix3x3.Multiply3x3(rasToLps, directionMatrix3x3, directionMatrix3x3)
+
+    # Create VTK image data
+    vtkImage = vtk.vtkImageData()
+    vtkImage.ShallowCopy(volumeNode.GetImageData())
+
+    # Update VTK image direction/origin/spacing
+    vtkImage.SetDirectionMatrix(directionMatrix3x3)
+    vtkImage.SetOrigin(volumeNode.GetOrigin())
+    vtkImage.SetSpacing(volumeNode.GetSpacing())
+
+    return itk.image_from_vtk_image(vtkImage)
+
+
+def itkImageFromVolumeModified(volumeNode):
+    """Indicate that modification of a ITK image returned by :py:meth:`itkImageFromVolume` (or
+    associated with a volume node using :py:meth:`updateVolumeFromITKImage`) has been completed."""
+    imageData = volumeNode.GetImageData()
+    pointData = imageData.GetPointData() if imageData else None
+    if pointData:
+        if pointData.GetScalars():
+            pointData.GetScalars().Modified()
+        if pointData.GetTensors():
+            pointData.GetTensors().Modified()
+        if pointData.GetVectors():
+            pointData.GetVectors().Modified()
+    volumeNode.Modified()
+
+
+def addVolumeFromITKImage(itkImage, name=None, nodeClassName=None):
+    """Create a new volume node from content of an ITK image and add it to the scene.
+
+    See :py:meth:`updateVolumeFromITKImage` to understand memory ownership.
+
+    :param itkImage: ITK image containing volume voxels.
+    :param name: volume node name
+    :param nodeClassName: type of created volume, default: ``vtkMRMLScalarVolumeNode``.
+      Use ``vtkMRMLLabelMapVolumeNode`` for labelmap volume, ``vtkMRMLVectorVolumeNode`` for vector volume.
+
+    :return: created new volume node
+    """
+    import slicer
+
+    if name is None:
+        name = ""
+    if nodeClassName is None:
+        nodeClassName = "vtkMRMLScalarVolumeNode"
+
+    volumeNode = slicer.mrmlScene.AddNewNodeByClass(nodeClassName, name)
+    updateVolumeFromITKImage(volumeNode, itkImage)
+    volumeNode.CreateDefaultDisplayNodes()
+
+    return volumeNode
+
+
+def updateVolumeFromITKImage(volumeNode, itkImage):
+    """Set voxels of a volume node from an ITK image.
+
+    .. warning:: Important: Memory area is shared between the ITK image and the ``vtkImageData``
+      container associated with the MRML volume node, therefore modifying the ITK image values
+      requires to call :py:meth:`itkImageFromVolumeModified`.
+
+      If the ITK image is reallocated, calling this function is required.
```

Comment:
slicer.util.updateVolumeFromArray uses deep-copy, so for consistency (and safety) it could be better to deep-copy the image by default. There could be an optional flag to enable shallow-copy.